tests/pipert/core/base/test_wires_validator.py

```diff
@@ -0,0 +1,160 @@
+import pytest
+from pytest_mock import MockerFixture
+from src.pipert2 import MiddleRoutine
+from src.pipert2 import Wire
+from src.pipert2 import DestinationRoutine
+from src.pipert2 import SourceRoutine
+from src.pipert2.core.base import wires_validator
+from src.pipert2.utils.exceptions.pipe_validation_failed import PipeValidationError
+
+
+@pytest.fixture()
+def valid_wires(mocker: MockerFixture):
+    source_routine = mocker.MagicMock(spec=SourceRoutine)
+    source_routine.name = "source_routine"
+
+    middle1_routine = mocker.MagicMock(spec=MiddleRoutine)
+    middle1_routine.name = "middle1_routine"
+
+    middle2_routine = mocker.MagicMock(spec=MiddleRoutine)
+    middle2_routine.name = "middle2_routine"
+
+    middle3_routine = mocker.MagicMock(spec=MiddleRoutine)
+    middle3_routine.name = "middle3_routine"
+
+    middle4_routine = mocker.MagicMock(spec=MiddleRoutine)
+    middle4_routine.name = "middle4_routine"
+
+    destination_routine = mocker.MagicMock(spec=DestinationRoutine)
+    destination_routine.name = "destination_routine"
+
+    wires = [
+        Wire(source=source_routine, destinations=(middle1_routine,)),
+        Wire(source=middle1_routine, destinations=(middle2_routine,)),
+        Wire(source=middle1_routine, destinations=(middle3_routine,)),
+        Wire(source=middle3_routine, destinations=(destination_routine,)),
+        Wire(source=middle2_routine, destinations=(destination_routine,)),
+        Wire(source=middle4_routine, destinations=(destination_routine,)),
+        Wire(source=middle2_routine, destinations=(middle4_routine,)),
+    ]
+
+    return wires
+
+
+def test_validate_existing_source_and_destination_routines(valid_wires):
+    try:
+        wires_validator.validate_existing_source_and_destination_routines(valid_wires)
+    except PipeValidationError as error:
+        pytest.fail(str(error))
```

Comment:
You can remove try except and just let the Exception raise.
If you decide to to keep this, maybe change the error message to, An error has raised...